### PR TITLE
Fix join_lines not adding a comma after join_single_expr_block with match arm

### LINKED ISSUE #868: Join lines should add `,` to match arm
See this test:

https://github.com/rust-analyzer/rust-analyzer/blob/2fa2805887e734647aabebc1f533ec76c48f538a/crates/ra_ide_api_light/src/join_lines.rs#L210-L245

Here, join lines produces invalid code because it doesn't add `,` after the block.

I think `join_single_expr_block` should be adjusted to add comma in such cases.

### PR DESCRIPTION
**Pull request by vipentti**
Fixes #868

### CODE REVIEW

**Comment by matklad** [comment]
>We will also remove optional whitespace between the expression and the comma.

I am not sure we really want this behavior *as a part of join_lines*. Removing space before comma seems like it really belongs to formater. I think if we restrict the feature to just adding the comma, the could becomes simpler? Basically, we'll need to append `,` to expression's text.

**Comment by vipentti** [comment]
Fair enough, will update. We'll still need to check for the presence of a comma after the original match_arm,  so we don't add an extra comma.

File: crates/ra_ide_api_light/src/join_lines.rs [hunk 1]
```diff
@@ -120,11 +120,35 @@ fn remove_newline(
...
+    let next = node.next_sibling();
+    let nnext = node.next_sibling().and_then(|n| n.next_sibling());
+
+    match (next, nnext) {
+        // Whitespace followed by a comma is fine
+        (Some(ws), Some(comma)) if ws.kind() == WHITESPACE && comma.kind() == COMMA => true,
+        (Some(n), _) => n.kind() == COMMA,
+        _ => false,
+    }
+}
```
Comment (matklad) [inline crates/ra_ide_api_light/src/join_lines.rs]: I've just remembered that we already have something similar:

https://github.com/rust-analyzer/rust-analyzer/blob/55da8e786d2cac3cb6332f80c1ee715fbcbdee0d/crates/ra_assists/src/lib.rs#L107

Let's move that function to `ra_syntax::algo` and use it here?

Comment (vipentti) [reply crates/ra_ide_api_light/src/join_lines.rs]: Sounds good, will do.

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**